### ISSUE TITLE
mgba froze at pokemon pinball bonus

### ISSUE THREAD

**Issue by sergiobenrocha2** [mgba-emu/mgba#510]
I was playing Pokemon Pinball - Ruby & Sapphire, when it was entering the bonus game it froze. Using perf top I can see high cpu usage in the mTimingDeschedule. Using last master commit, libretro core

![mgba - froze and high cpu usage](https://cloud.githubusercontent.com/assets/5431285/22516623/7e15fdce-e88d-11e6-8542-758c20d651d6.png)

**Comment by endrift** [comment]
This means that something got scheduled twice, without it being descheduled first...hm. I wonder what.

**Comment by sergiobenrocha2** [comment]
```
Program received signal SIGINT, Interrupt.
0x00007fffec027108 in mTimingDeschedule (timing=timing@entry=0x7ffff7ff66c0, event=event@entry=0x7ffff7ff5a98) at /home/sergio/Projects/mGBA/mgba-git/src/core/timing.c:48
48			if (next == event) {
(gdb) info threads
  Id   Target Id         Frame 
  8    Thread 0x7fffe920f700 (LWP 18434) "threaded-ml" 0x00007ffff268afdd in poll () at ../sysdeps/unix/syscall-template.S:81
  4    Thread 0x7fffd7b2f700 (LWP 18189) "retroarch" __lll_lock_wait () at ../nptl/sysdeps/unix/sysv/linux/x86_64/lowlevellock.S:135
  2    Thread 0x7fffecaea700 (LWP 18159) "retroarch" pthread_cond_wait@@GLIBC_2.3.2 () at ../nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_wait.S:185
* 1    Thread 0x7ffff7e23780 (LWP 18155) "retroarch" 0x00007fffec027108 in mTimingDeschedule (timing=timing@entry=0x7ffff7ff66c0, event=event@entry=0x7ffff7ff5a98)
    at /home/sergio/Projects/mGBA/mgba-git/src/core/timing.c:48
(gdb) thread 1
[Switching to thread 1 (Thread 0x7ffff7e23780 (LWP 18155))]
#0  0x00007fffec027108 in mTimingDeschedule (timing=timing@entry=0x7ffff7ff66c0, event=event@entry=0x7ffff7ff5a98) at /home/sergio/Projects/mGBA/mgba-git/src/core/timing.c:48
48			if (next == event) {
(gdb) print timing->root->name
$1 = 0x7fffec0a91f7 "GBA Audio Sample"

```

**Comment by sergiobenrocha2** [comment]
```
(gdb) bt full
#0  0x00007fffec027108 in mTimingDeschedule (timing=timing@entry=0x7ffff7ff66c0, event=event@entry=0x7ffff7ff5a98) at /home/sergio/Projects/mGBA/mgba-git/src/core/timing.c:48
        previous = 0x7ffff7ff54b8
        next = 0x7ffff7ff5498
#1  0x00007fffec08b852 in GBADMAUpdate (gba=gba@entry=0x7ffff7ff5000) at /home/sergio/Projects/mGBA/mgba-git/src/gba/dma.c:189
        i = <optimized out>
        memory = 0x7ffff7ff5020
        currentTime = 4079153830
#2  0x00007fffec08bd0f in GBADMASchedule (gba=gba@entry=0x7ffff7ff5000, number=number@entry=3, info=info@entry=0x7ffff7ff5a74)
    at /home/sergio/Projects/mGBA/mgba-git/src/gba/dma.c:115
No locals.
#3  0x00007fffec08bdb6 in GBADMAWriteCNT_HI (gba=gba@entry=0x7ffff7ff5000, dma=dma@entry=3, control=<optimized out>, control@entry=33024)
    at /home/sergio/Projects/mGBA/mgba-git/src/gba/dma.c:85
        memory = 0x7ffff7ff5020
        currentDma = 0x7ffff7ff5a74
        wasEnabled = 0
#4  0x00007fffec079d30 in GBAIOWrite (gba=gba@entry=0x7ffff7ff5000, address=address@entry=222, value=value@entry=33024) at /home/sergio/Projects/mGBA/mgba-git/src/gba/io.c:485
No locals.
#5  0x00007fffec07950d in GBAIOWrite32 (gba=gba@entry=0x7ffff7ff5000, address=address@entry=220, value=value@entry=2164310016)
    at /home/sergio/Projects/mGBA/mgba-git/src/gba/io.c:656
No locals.
#6  0x00007fffec07eb79 in GBAStore32 (cpu=0x7ffff7ff7000, address=67109084, value=-2130657280, cycleCounter=0x7fffffffdb9c)
    at /home/sergio/Projects/mGBA/mgba-git/src/gba/memory.c:735
        gba = 0x7ffff7ff5000
        memory = 0x7ffff7ff5020
        wait = 0
        waitstatesRegion = 0x7ffff7ff57c4 ""
#7  0x00007fffec03fbee in _ThumbInstructionSTR1 (cpu=0x7ffff7ff7000, opcode=<optimized out>) at /home/sergio/Projects/mGBA/mgba-git/src/arm/isa-thumb.c:100
        currentCycles = 2
        immediate = <optimized out>
        rd = <optimized out>
        rm = <optimized out>
#8  0x00007fffec044351 in ThumbStep (cpu=0x7ffff7ff7000) at /home/sergio/Projects/mGBA/mgba-git/src/arm/arm.c:282
        opcode = <optimized out>
#9  ARMRunLoop (cpu=0x7ffff7ff7000) at /home/sergio/Projects/mGBA/mgba-git/src/arm/arm.c:299
No locals.
#10 0x00007fffec07bd41 in _GBACoreRunFrame (core=0x8ad0c0) at /home/sergio/Projects/mGBA/mgba-git/src/gba/core.c:335
        gba = 0x7ffff7ff5000
        frameCounter = 29811
#11 0x00007fffec0a4f90 in retro_run () at /home/sergio/Projects/mGBA/mgba-git/src/platform/libretro/libretro.c:269
        keys = <optimized out>
        var = {key = 0x7fffec0af600 "mgba_allow_opposing_directions", value = 0x0}
        updated = false
        wasAdjustingLux = false
        width = 4458642
        height = 0
#12 0x0000000000416071 in core_run () at core_impl.c:437
No locals.
#13 0x0000000000426221 in runloop_iterate (sleep_ms=sleep_ms@entry=0x7fffffffdce0) at runloop.c:1156
        i = <optimized out>
        current = <optimized out>
        target = <optimized out>
        to_sleep_ms = <optimized out>
        trigger_input = 0
        last_input = 0
        input_driver_is_nonblock = false
        settings = 0x7fffecaeb010
        old_input = <optimized out>
        menu_is_alive = <optimized out>
        current_input = <optimized out>
#14 0x000000000041500a in rarch_main (argc=4, argv=<optimized out>, data=0x0) at frontend/frontend.c:130
        sleep_ms = 0
        ret = <optimized out>
        args = 0x0
#15 0x00007ffff25bff45 in __libc_start_main (main=0x4124a0 <main>, argc=4, argv=0x7fffffffde08, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, 
    stack_end=0x7fffffffddf8) at libc-start.c:287
        result = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {0, 5474901377060021565, 4269223, 140737488346624, 0, 0, -5474901378040748739, -5474928555579169475}, mask_was_saved = 0}}, 
          priv = {pad = {0x0, 0x0, 0x52aac0 <__libc_csu_init>, 0x7fffffffde08}, data = {prev = 0x0, cleanup = 0x0, canceltype = 5417664}}}
        not_first_call = <optimized out>
#16 0x00000000004124d0 in _start ()
No symbol table info available.
```

**Comment by endrift** [comment]
Should be fixed in 181c05c7ac8a739cd6d9dea393ac9a30e1898d96. Thanks for finding this!
